### ISSUE TITLE
Decide what to do with Smart TV Browser apps (Testing, Support)

### ISSUE THREAD

**Issue by metal-messiah** [newrelic/newrelic-browser-agent#121]
[NOTE]: # ( ^^ Provide a general summary of the request in the title above. ^^ )

## Summary

[NOTE]: # ( Provide a brief overview of what the new feature is all about. )

We are seeing more and more issues pop up with Smart TV apps.  Smart TV will also only grow from here.  We have no testing , focus, or plan in place for any Smart TV environment.  This ticket could serve to get a discussion going about how we want to approach Smart TV apps moving forward

## Possible Solution

[NOTE]: # ( Not required. Suggest how to implement the addition or change. )
Some options:
- [ ] Introduce tests for Smart TV environments
- [ ] Plan around smart TV environments (TV apps are almost entirely streaming focused, is there implementations we could make to better support streaming?)
- [ ] Decide what Smart TV environments (if any) we support
- [ ] Have platform specific tool for TV (built more around video?)
- [ ] Do nothing

## Additional context

[TIP]:  # ( Why does this feature matter to you? What unique circumstances do you have? )
Smart TV app development will only grow as more people have Smart TVs than ever before, and many companies are embracing a streaming video approach from home over other mediums.  We should at least consider our options for staying relevant with these apps and try to mitigate issues if we can.

**Comment by aubreymasten** [comment]
In the past we have attempted to abort the agent by:
* Setting a timeout after requesting the second half of the agent
* Adding an event listener to the JSONP call to identify failed network requests

We could instrument these methods with a supportability metric to track how often the agent fails to load. 

In the case where we aren't able to successfully abort the agent, we may need to identify other strategies for aborting the agent. It's also possible our current detection approaches are working, but we're not correctly aborting the agent and there are still memory leaks to be found.

**Comment by cjmccarthy** [comment]
@metal-messiah @aubreymasten We have [a lot of trackers](https://github.com/newrelic-experimental/video-documentation/wiki/Player-Support-by-Platform) for streaming video built by @asllop that our Media customers are using in production for web environments - Smart TVs and Game Consoles included. It can be an adventure with those customers to get the browser agent working correctly in those environments, but I can confirm there is a huge use case here and more official support for Smart TV and Console environments could help us drive a lot of usage.

**Comment by asllop** [comment]
Hi @metal-messiah ! NR's main video telemetry engineer here. I have successfully integrated the browser agent in Chromecast, but haven't tested any other JS-based SmartTV. Actually, I have been told that some Smart TVs (especially old ones) can't run the browser agent, and this let me modify the video agent core to accept [custom event loggers](https://github.com/newrelic/video-core-js/blob/c437dac0172e6e709b445f4d15a0dc060c433592/src/core.js#L85) besides the NR browser agent. Regarding the subset of SmartTVs that should be supported, have a look at this [market share investigation](https://docs.google.com/document/d/1cMghdJrTxGzjmVRcw9IZE0sUdaj_XTJKDzuYynbh7xM/edit?usp=sharing) I made a few months ago: The main ones worldwide are Samsung (Tizen) and Smart Alliance based systems. If you feel that need my collaboration just give me a shout.

cc @cjmccarthy
